src/stream/stream.rs

```diff
@@ -21,10 +21,12 @@
 //! # }) }
 //! ```
 
+use std::cmp::Ordering;
 use std::pin::Pin;
 
 use cfg_if::cfg_if;
 
+use super::min_by::MinBy;
```

Comment:
Since `min_by` should ideally be an `async fn`, let's make `MinBy` a hidden type and not even implement `Debug` (because `async fn` desugaring doesn't implement `Debug` for its future).